javascript binary search tree: Add Additional resources (#25487)
TheOdinProject/curriculum: javascript/computer_science/project_binary_search_trees.md

<div class="lesson-content__panel" markdown="1">

1.  Build a `Node` class / factory. It should have an attribute for the data it stores as well as its left and right children.

1.  Build a `Tree` class / factory which accepts an array when initialized. The `Tree` class should have a `root` attribute which uses the return value of `buildTree` which you'll write next.

1.  Write a `buildTree` function which takes an array of data (e.g. [1, 7, 4, 23, 8, 9, 4, 3, 5, 7, 9, 67, 6345, 324]) and turns it into a balanced binary tree full of `Node` objects appropriately placed (don't forget to sort and remove duplicates!). The `buildTree` function should return the level-0 root node.

    **Tip:** If you would like to visualize your binary search tree, here is a `prettyPrint()` function that will `console.log` your tree in a structured format. This function will expect to receive the root of your tree as the value for the `node` parameter.

>     ```javascript

Reviewer: `~~~` would be in accordance to TOP's style guide here.

Reviewer: @taladan - You can commit and push to your remote branch. It'll get reflected.

Reviewer: And yes, VSCode's auto-save formatting defaults to backticks for fenced codeblocks.

1. Print out all elements in level, pre, post, and in order

</div>

### Additional Resources

This section contains helpful links to related content. It isn't required, so consider it supplemental.

If you need more help on the three types of binary search tree traversal algorithms, these video resources may aid you:

> - [Inorder Traversal Algorithm](https://www.youtube.com/watch?v=4_UDUj1j1KQ&t=1s) of a binary search tree by Yicheng Gong

Reviewer: I like the way this is added here, but could you explain why and when a learner should watch these? Something in the lines of "this helps in further visualising how a call stack looks when traversing the tree" should be cool enough IMO.

Author: This has been implemented

> If you need more help on the three types of binary search tree traversal algorithms, these video resources may aid you in visualizing how a call stack looks when traversing the tree:
> 
> - [Inorder Traversal Algorithm](https://www.youtube.com/watch?v=4_UDUj1j1KQ&t=1s) of a binary search tree by Yicheng Gong
> - [Post-order Traversal Algorithm](https://www.youtube.com/watch?v=4Xo-GtBiQN0) of a binary search tree by Yicheng Gong
> - [Pre-order Traversal Algorithm](https://www.youtube.com/watch?v=8xue-ZBlTKQ&ab_channel=ygongcode) of a binary search tree by Yicheng Gong

Reviewer 2: This does not follow our layout style guide for Additional Resources. The explanation before 3 bullet points do not allow for additional bullet points at a later time. 
https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md

In addition, the 3 bullet points are repetitive, so I think we should try and condense it down to one bullet point that contains all 3 links. Something like:

```suggestion
- Yicheng Gong has some excellent videos that help visualize the call stack when traversing binary search trees: [Inorder Traversal Algorithm](https://www.youtube.com/watch?v=4_UDUj1j1KQ&t=1s), [Post-order Traversal Algorithm](https://www.youtube.com/watch?v=4Xo-GtBiQN0), and [Pre-order Traversal Algorithm](https://www.youtube.com/watch?v=8xue-ZBlTKQ). 
```

Reviewer 2: We might be able to condense the links even more, something like:
```suggestion
- Yicheng Gong has some excellent videos that help visualize the call stack when traversing binary search trees: [Inorder](https://www.youtube.com/watch?v=4_UDUj1j1KQ&t=1s), [Post-order](https://www.youtube.com/watch?v=4Xo-GtBiQN0), and [Pre-order](https://www.youtube.com/watch?v=8xue-ZBlTKQ) Traversal Algorithms.
```
